README.md

```diff
@@ -1 +1,123 @@
-# PackageDistroTools
\ No newline at end of file
+# PackageDistroTools
+
+This repository contains tools for maintaining the
+[GAP package distribution repository](https://github.com/gap-system/PackageDistro).
+
+## The scripts
+
+The following scrips are provided. The usage examples assume that git clones
+of `PackageDistroTools` and `PackageDistro` live next to each other under a
+common parent directory.
+
+### `scan-for-updates.py`
+
+
+Iterates over all metadata files in the package distribution, and do the following:
+
+- extract the `PackageInfoURL` and `PackageInfoSHA256` fields from the
+  metadata
+- download the `PackageInfo.g` file from the URL, find the sha256 of the
+  file, and compare it to the one in the metadata.
+- if the sha256 hashes differ, then write the downloaded `PackageInfo.g`
+  file to `_tmp/package_name.g`
+
+Usage:
+
+    cd ../PackageDistro
+    ../PackageDistroTools/scan-for-updates.py
+
+
+### `import-package.py`
+
+Takes one (or multiple?) PackageInfo.g file (path) as argument, then...
+  - validates it using `ValidatePackageInfo()` 
+  - reject `DEV` package versions
+  - reject versions (and release dates?) that come before the package info currently in store for that package (if any)
+  - possibly do further validation (e.g. validate `README_URL`???)
+  - downloads one of the associated archives (and perhaps extracts its `PackageInfo.g`, and verifies it has a matchin checksum?)
+    - if possible we want .tar.gz; in case any package doesn't have one, we may need a script to generate one (e.g. tar.bz2 is trivial, not sure if there is a package that only offers .zip? perhaps we should *require* a `.tar.gz`???)
+  - computes checksums of PI.g and the archive, adds them to the metadata
+  - writes out the metadata as JSON
+- alternate usecase: take the name of a .tar.gz as argument;
+  extract its `PackageInfo.g`, then proceed similar to before?!?
+
+
+### `upload-package.py`
+- to upload such a package tarball to GitHub release (could use [GitHub's `gh` tool](http://cli.github.com) or adapt the code)
+- also would be nice to be able to upload to our own server, but that'll probably amount to a `scp FILE.tar.gz ourserver:/some/path` done in a GH action, which gets a suitable SSH prviate key (granting it write access) via a repository secreet
+
+
+### `assemble-distro.py`
+- scans over all metadata, collects the appropriate tarballs, then "merges" them into one big tarball
+- the assembled distro archive also needs to be uploaded, but ideally we can just repurpose `upload-package.py`
+- should do caching: download tarballs into a specific fixed location; if a file with the right name already exists and has the right checksum, just use it
```

Comment:
So far my naive thought was that this script downloads the ~150 tarballs each time and assembles them.

But actually it's perhaps better to just download the previous package-archive.tar.gz, extract it, remove the old package(s), insert any new or update package(s), then compress again and upload. My only concern is race conditions: if this step fails for some reason at some point, we may end up with a mismatch between the content of that archive and what the metadata in repository says should be... HRM.

So perhaps still better to always download all the package tarballs -- that approach probably also is better for caching purposes...